packages/compose/src/hooks/use-previous/index.js

```diff
@@ -24,7 +26,7 @@ export default function usePrevious( value ) {
 	// Store current value in ref.
 	useEffect( () => {
 		ref.current = value;
-	}, [ value ] ); // Re-run when value changes.
+	}, optionalDependencyList ); // Re-run when any dependency changes.
```

Comment:
This could be error prone, if the length of the dependency list changes, then it will throw unexpected errors.

Also, this could violates the official-recommended eslint rules, which means could introduce errors that's not easy to debug.

Reply:
how could the length of the dependency list change here?

Reply:
Though unlikely, but...

```js
const [deps, setDeps] = useState([]);
usePrevious(_, deps);
setDeps([1]);
```

I would suggest to almost always avoid adding dependency list as an argument for a custom hook. As it's error prone and cannot be easily spotted if there's an error. In this specific case, I would probably just inline the implementation. As I can't think of any other place we would need this to be a general utility hook.